datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -2316,6 +2337,19 @@ mod test {
             Field::new_list_field(DataType::Int32, true).into(),
         ))?;
 
+        round_trip_type(DataType::Map(
+            Field::new_struct(
```

Comment:
Maybe we can also add a test using `Field::new_map`: https://docs.rs/arrow/latest/arrow/datatypes/struct.Field.html#method.new_map

we could do it in a follow on PR too

Reply:
Field::new_map is a bit annoying as it creates a map field directly, rather than the inner field of a map.. would be nice to have something like `new_list_field` but for map, but Arrow doesn't have it.

Anyways I could replace this block or add another one with 
```
        round_trip_type(
            Field::new_map(
                "map",
                "entries",
                Field::new("key", DataType::Utf8, false).into(),
                Field::new("value", DataType::Int32, true).into(),
                false,
                false,
            )
            .data_type()
            .to_owned(),
        )?;
```
frankly I don't know if it adds much value, but I'm fine either way - do you have a preference?

Reply:
If you don't think it is valuable let's not use it